lib/openshift_client.rb

```diff
@@ -20,17 +20,50 @@ class Client < Kubeclient::Common::Client
       [OpenshiftClient.const_set(et, Class.new(RecursiveOpenStruct)), et]
     end
 
-    def initialize(uri, version = 'v1', path = '/oapi')
+    def initialize(uri,
+                   version = 'v1',
+                   path = '/oapi',
+                   ssl_options: {
+                     client_cert: nil,
+                     client_key: nil,
+                     ca_file: nil,
+                     verify_ssl: OpenSSL::SSL::VERIFY_PEER
+                   },
+                   auth_options: {
+                     user: nil,
+                     password: nil,
+                     bearer_token: nil
+                   })
+      fail ArgumentError, 'Missing uri' if uri.nil?
+
+      validate_auth_options(auth_options)
+
       handle_uri(uri, path)
       @api_version = version
       @headers = {}
-      ssl_options
+      @ssl_options = ssl_options
+
+      @basic_auth_user = auth_options[:user]
+      @basic_auth_password = auth_options[:password]
+      bearer_token(auth_options[:bearer_token]) if auth_options[:bearer_token]
     end
 
     def all_entities
       retrieve_all_entities(ENTITY_TYPES)
     end
 
+    private
+
+    def validate_auth_options(opts)
```

Comment:
@moolitayer any way to reuse the kubeclient `validate_auth_options`? (instead of copy/paste)